src/common.rs

```diff
@@ -60,7 +60,13 @@ pub fn get_bytes<'a>(py: Python, bytes: &'a PyObject) -> Option<&'a [u8]> {
     } else if let Ok(bytes) = PyString::try_from(bytes.as_ref(py)) {
         Some(bytes.as_bytes())
     } else if let Ok(bytes) = PyByteArray::try_from(bytes.as_ref(py)) {
-        Some(bytes.data())
+        let v = bytes.to_vec();
```

Comment:
I think a better way would be to have:
```rust
pub(crate) enum Bytes<'a> {
    Slice(&'a [u8]),
    Vec(Vec<u8>),
}
```
and return Option<Bytes<'a>> and then handle the 2 cases in src/init.rs

Reply:
Done with ccc0a10.